### PR TITLE
fix: rework audit progress calculation

### PR DESCRIPTION
**Pull request by Mohamed-Hacene**
## What & why

Audits showed 100% progress without being filled in when global scoring was enabled: enabling scoring pre-fills every score at the scale minimum, and the old rule (`result set OR score present`) counted those pre-filled scores as assessed.

Progress is now computed by a single rule, shared by every surface (tables, audit page, journeys, My assignments):

- **Status field enabled** on the audit: a requirement counts once it is **Done**, nothing else.
- **Status hidden**: questionnaire fully answered, else result set, else score strictly above the minimum.

## Perf

| Page / endpoint | Old (SQLite / PG) | New (SQLite / PG) | Why |
|---|---|---|---|
| **Audits table: hidden per-page prefetch (REMOVED)** | **673 / 654 ms** | 0 | The old list action prefetched every assessable requirement (+ its node) of the page's audits — ~20k ORM objects hydrated per page load — solely to serve the old IG progress path. Nothing reads them anymore; the block is deleted. This dwarfs every other number in this table. |
| Audits table GROUP BY, 20 audits / 20k req., status mode (default) | 6.0 / 7.0 ms, 1 query | 5.8 / 6.6 ms, 3 queries | No measurable regression. Two extra metadata lookups bounded by the page; the main GROUP BY is a plain `status = 'done'` filter, simpler than the old OR condition. |
| Audits table, content mode (result visible) | 6.0 / 7.0 ms | 6.2 / 6.9 ms | No measurable regression: compiles to the same query shape as the old one. |
| Audits table, mixed page (20 classic + 5 questionnaire audits) | 6.1 / 6.5 ms | 8.8 / 11.0 ms | +2.7 to +4.5 ms paid only by the questionnaire audits' bucket: three indexed `EXISTS` per requirement of those 5 audits (has questions, has seeded answers, has unanswered answers — the last two let fully answered questionnaires count even when their questions compute neither result nor score). |
| Audits table, 5 IG audits (5k req.) | heavy prefetch + per-audit Python | 13.4 / 81.2 ms, 3 queries for the whole page | One shared scalar `.values()` scan for all IG audits of the page (IG membership intersects two JSON lists, undoable in SQL). PG pays per-row fetch overhead over TCP; bounded by the page's IG audits, which are a minority. |
| `progress` field of `GET /compliance-assessments/{id}` (also feeds X-Rays and the analytics widget), 1k req. | 1.3 / 1.2 ms | **0.7 / 1.8 ms** (content mode; 1.3 / 4.4 ms on a questionnaire framework; status mode is cheaper still) | Old: two `Count(distinct=True)` over the audit→requirements→nodes LEFT JOIN, attached to every detail action. New: one direct aggregate, only where the field is serialized. |
| My assignments (auditee dashboard), questionnaire audit (200 RAs / 600 questions, one assignment covering all) | 49.7 / 66.8 ms, 7 queries | **43 / 60 ms, 5 queries (constant)** | Kept on the Python `get_visible_questions_counts()` walk (visibility-aware), identical to `main` — neutral, not a regression. An intermediate SQL-aggregate version measured 1.6 / 12.8 ms but was dropped: it couldn't resolve conditional visibility, producing the informational+conditional gap and diverging from the assessment page. The ~40 ms is the price of correctness; bounded to the respondent's own (few) assignments, N+1 closed (one batched `Prefetch`, constant query count). Pagination is the intended scaling lever, not further per-page optimization. |

### Scaling ceilings

- Every query is filtered by the page's audit ids (indexed), so cost scales
  with **the requirements of the audits displayed**, never with total table
  size. Ceiling = page size x largest framework: 20 audits x 5,000
  requirements ≈ 100k rows ≈ ~30 ms extrapolated from the measurements.
- Buckets are keyed by configuration, not by audit: (status mode) + (result
  visible x has questions) = **5 buckets maximum**, hence a hard cap of
  **7 queries per page** (5 GROUP BY + 2 metadata lookups); 3 in practice.
  The buckets partition the same row volume a single query would scan.
- The answers table — the only one growing unboundedly with activity — is
  never read by any progress path.
- Weakest spot: the shared IG scan (~13 ms SQLite / ~81 ms PostgreSQL for
  5,000 rows, per-row fetch cost). One scan per page whatever the number of
  IG audits, bounded by their framework sizes, and only paid when the page
  contains IG audits.

## Test plan

- Reproduced the bug (audit with pre-filled scores): 100% on main, 0% with the patch.
- Verified both modes and list/detail consistency on a dev database; benchmarked old vs new on a seeded 20k-requirement dataset.

## Summary

- **Improvements**
  - Progress tracking is now consistently driven by questionnaire/answer state and per-field visibility (status vs content) across journeys, lists, and dashboards.
  - Scoring rules are tightened: question-based scoring requires fully answered visible questionnaires, and provided score inputs are ignored when questions are present.
  - List progress is computed more efficiently, and progress/statistics endpoints use page-scoped, visibility-aware counting.
  - Respondent-alignment-based completion is supported with updated ToC section/ToC dot coloring.

- **Tests**
  - Expanded coverage for status vs content visibility modes, score-only edge cases, and implementation-group filtering.

### CODE REVIEW

**Comment by ab-smith** [comment]
@coderabbitai full review

File: backend/core/models.py [hunk 1]
```diff
@@ -8829,6 +8873,124 @@ def trigger_compliance_assessment_update_hooks(self):
...
+                F("compliance_assessment__min_score"),
+                F("compliance_assessment__framework__min_score"),
+                Value(0),
+            )
+        )
+        content = result_assessed if result_visible else score_progressed
+        if has_questions_annotation:
+            content = (Q(_has_questions=True) & (result_assessed | score_assessed)) | (
+                Q(_has_questions=False) & content
+            )
```
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply backend/core/models.py]: Intentional: completion is carried by existing fields (recompute resolves question visibility at write time) plus the all-answers-non-empty signal. A persisted completion flag was considered and rejected (no schema change for a derivable value). Remaining edges are documented; validated on real data. Won't fix here.

File: frontend/src/routes/(app)/(third-party)/auditee-assessments/[id=uuid]/+page.svelte [hunk 2]
```diff
@@ -310,11 +310,30 @@
...
+	// Per-item completion, mirroring the backend cascade: a question-bearing
+	// requirement completes through its answers (whether or not they compute
+	// a result), alignment-driven ones through the respondent's alignment
+	// answer, the rest through the result.
+	function isItemDone(item: { data: Record<string, any> }): boolean {
+		const visible = item.data.visible_questions ?? 0;
+		if (visible > 0) return (item.data.answered_questions ?? 0) >= visible;
+		if (useAlignmentProgress) return Boolean(item.data.respondent_alignment);
+		return item.data.result !== 'not_assessed';
+	}
```
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply frontend/src/routes/(app)/(third-party)/auditee-assessments/[id=uuid]/+page.svelte]: Actually the comment is wrong, this page is respondent-facing, so it tracks the respondent's own work (questions, alignment, result) and deliberately does not follow the audit-level progress mode

**Review by eric-intuitem — APPROVED** [review]
LGTM.
We'll need to add a db field to enable sorting, separate PR.
